pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py

```diff
@@ -18,16 +18,24 @@ def _VSCODE_convertNumpyArrayToDataFrame(ndarray):
     try:
         x_len = temp.shape[0]
         # Figure out if we're dealing with ragged data
-        # Handle ragged arrays by making a container where the number of
-        # columns is the max length of all rows. Missing elements are
-        # represented as empty strings.
         row_lengths = [
             len(temp[i])
             if isinstance(temp[i], (list, tuple, _VSCODE_np.ndarray))
             else 1
             for i in range(x_len)
         ]
-        flattened = _VSCODE_np.full((x_len, max(row_lengths)), "", dtype="object")
+        max_y_len = max(row_lengths)
+        is_ragged_or_ndimensional = hasattr(temp, "ndim") and (
```

Comment:
Ok, this is probably just explaining python to a python noob. But is there a missing case here in the case where ndim == 2, but it also has a ragged type?

Reply:
Looking on the ndarray docs now

Reply:
I was originally thinking that that case doesn't matter because the elements along the 3rd dimension get stringified, which is what we would've done anyway in the special handling. But FWIW the stringification for those nested array elements looks like this:

![image](https://user-images.githubusercontent.com/30305945/106969126-17fbbe80-66ff-11eb-830f-ebe60cc1b20c.png)

Maybe I shouldn't be handling 1D ragged ndarrays specially, i.e. for this data:
```python
bar = np.array([[1, 2, 3, 4], [5, 6, 7]])
```

instead of presenting them nicely like this (current special-cased behavior):
![image](https://user-images.githubusercontent.com/30305945/106969469-b5ef8900-66ff-11eb-9279-5d7b35c43eac.png)

we would instead present them like this: 
![image](https://user-images.githubusercontent.com/30305945/106969555-e59e9100-66ff-11eb-85e7-a75fb74cc86c.png)

Reply:
This is honestly beyond my user knowledge on what a data scientist would want. Personally I think your current non-special case scenario looks like what I would want. I see that as a 1D nparray which contains two things that happen to be lists. But I do not know if the general community would agree.

Reply:
Is there prior art here from spyder / pycharm / r studio?

Reply:
Serves me right for trying to be too clever with handling ragged arrays 😁 thanks for bringing this up Ian. Much happier with the code as well, and this way we're more consistent with other DS tooling as well.